### PR TITLE
Testlib request methods

### PR DESCRIPTION
**Pull request by will-moore**
# What this PR does

Cleanup of ```omeroweb.testlib``` Django client request methods, based on https://trello.com/c/flvJ728g/31-web-api-tests-to-include-csrf.

Previously we've had a number of ```testlib``` methods such as:
```
_post_response
_post_response_json
_csrf_post_response
_csrf_post_response_json
_csrf_post_json
_csrf_put_json
_delete_response
_csrf_delete_response
_csrf_delete_response_json
_get_response
_csrf_get_response
_get_response_json
```
These try to cover a matrix of:
 - GET, POST, PUT, DELETE
 - With or with CSRF tokens
 - POST data as query string or JSON content
 - Handling response as text or JSON

These are replaced by a smaller number of methods.
For all POST, PUT & DELETE we first test *without* CSRF (check for 403 response) then *with* CSRF.

(This means that all the ```test_api_* ``` tests are now being tested for CSRF checks, as per trello card above).

```
get: GET without CSRF (parameter for CSRF option)
get_json: as above, but handles JSON response
post: POST - data encoded in query string
post_json: POST, encodes data as JSON and handles JSON response
put_json: PUT, encodes data as JSON and handles JSON response
delete_json: DELETE, encodes data as JSON and handles JSON response

csrf_response: underlying helper method is available if you want to do POST/PUT/DELETE without doing the CSRF test.

All ```Omeroweb/test/integration``` tests have been updated to use new methods.
Original methods above are marked with deprecation warnings.
```

# Testing this PR

1. Tests should be green.

### CODE REVIEW

File: components/tools/OmeroWeb/omeroweb/testlib/__init__.py [hunk 1]
```diff
@@ -110,16 +111,80 @@ def _response(django_client, request_url, method, data, status_code=403,
...
 
+def csrf_response(django_client, request_url, method, data=None,
+                  status_code=200, content_type=MULTIPART_CONTENT,
+                  test_csrf_required=True):
+    """
+    Helper for testing post/put/delete with and without CSRF.
+
+    :param django_client:   Django test Client
+    :param request_url:     The url to request
+    :param data:            A dict of data to include as json content
```
Comment (jburel) [inline components/tools/OmeroWeb/omeroweb/testlib/__init__.py]: missing ``method`` argument

File: components/tools/OmeroWeb/omeroweb/testlib/__init__.py [hunk 2]
```diff
@@ -110,16 +111,80 @@ def _response(django_client, request_url, method, data, status_code=403,
...
+    rsp = post(django_client, request_url, json.dumps(data),
+               status_code=status_code, content_type='application/json')
+    assert rsp.get('Content-Type') == 'application/json'
+    return json.loads(rsp.content)
+
+
 def _post_response(django_client, request_url, data, status_code=403,
                    content_type=MULTIPART_CONTENT, **extra):
+    warnings.warn(
+        "This method is deprecated as of OMERO 5.4.0", DeprecationWarning)
```
Comment (jburel) [inline components/tools/OmeroWeb/omeroweb/testlib/__init__.py]: Maybe indicate the one to use in the message

**Comment by mtbc** [comment]
Would you mind temporarily excluding this so we can try to get #5349 in?

**Comment by will-moore** [comment]
@mtbc Added exclude label to description.

**Comment by jburel** [comment]
@will-moore could you remove the exclude label and fix the merge conflict
now that #5349 is in

**Comment by will-moore** [comment]
Travis failure: "This job ran on our Trusty environment, which is gradually becoming our default Linux environment. You can add dist: precise in your .travis.yml file to continue using Precise."

**Comment by jburel** [comment]
@will-moore could you roll back your latest commit?
I am adjusting the travis file in gh-5409
so we can update pytables (see gh-5407)

**Comment by jburel** [comment]
Thanks for updating message
Tests are passing
Merging so we can unlock other PRs

**Pull request merged**